cypress/e2e/undo-redo/block-configs-should-update-flow-state/common-block-configs.cy.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * In general, e2e should focus on what are VISIBLE from users perspective.
+ * But, as we have a bunch of configs to test, we can assume that:
+ * - if a block config UI updates the flow state, it should work with the UNDO/REDO feature
+ *
+ * Secondly, a working config UI doesn't mean the component is working with UNDO/REDO, it has to mutate the flow state and not use local state.
+ */
+describe('mutate flow state when the update comes from common block configs UI', () => {
+  beforeEach(() => {
+    cy.createFlow({
+      label: 'adding blocks test',
+    })
+  })
+
+  it('should validate non interactive blocks', () => {
+    const blockConfigs = {
+      label: 'my block',
+      name: {
+        beforeEdit: 'my_block',
+        afterEditPostFix: '_updated',
+      },
+      resources: {
+        ivr: ' 02_flowers_for_albert.mp3',
+        sms: 'content for sms',
+        ussd: 'content for ussd',
+      },
+      exits: [
+        {
+          name: 'banana',
+          expression: '@block.value = "banana"',
+        }
+      ],
+      tags: [
+        {
+          name: 'my tag 1',
+        }
+      ]
+    }
+    const getStore = () => cy.window().its('store')
+    const flowModuleState = () => getStore().its('state.flow')
+    const flowsListState = () => flowModuleState().its('flows')
+
+    cy.addBlock(['Content', 'Message'])
+    const firstBlockState = () => flowsListState().its('[0].blocks[0]')
+    const firstResourcesState = () => flowsListState().its('[0].resources[0]')
+
+    // ####### Block label & name
+    cy.get('[data-cy="label--editor"]')
+      .find('textarea')
+      .type(String(blockConfigs.label))
```

Comment:
Will check if we do need to cast with `String()`, I will remove if not needed. (applied for many typing like this)